framework/behaviors/SluggableBehavior.php

```diff
@@ -171,6 +171,10 @@ protected function isNewSlugNeeded()
             return false;
         }
 
+        if ($this->attribute === null) {
```

Comment:
We want to know 'what' the software does and not 'how' he does it. 

I tried to get to this, but I could not get clear answers to my questions. Examples:

* My question "Can you support this with a comment to explain why true is to be returned when attribute is null" in https://github.com/yiisoft/yii2/pull/13729#discussion_r106017513 gets a vague implementation reaction in https://github.com/yiisoft/yii2/pull/13729#discussion_r106027562
* My question "what would be the usecase for that" gets an answer about implemantation details in https://github.com/yiisoft/yii2/pull/13729#discussion_r106033797 but not about the "why" which was my question
* My last question to understand the "why" https://github.com/yiisoft/yii2/pull/13729#discussion_r106035189 was about trying to get you to formulate the usecase into a understandable testMethod, but again I dont get an corresponding answer to my question in https://github.com/yiisoft/yii2/pull/13729#discussion_r106101040

To go back to the code. My problem is to understand the logic behind `$this->attribute === null`. What is the meaning of it. Why is it added? By purely looking at the code, I have to put effort into understand it. This goes against the best practice that code should be self-explanatory and instantly understandable. 

So again. We want to know 'what' the software does and not 'how' he does it.